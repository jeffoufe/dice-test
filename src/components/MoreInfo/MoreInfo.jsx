// @flow

import React, { useState } from 'react';
import { ImportantText, Text } from '../../components/Typography/Typography';
import List from '../../components/List/List';
import styles from './MoreInfo.module.scss';
import buildPrice from '../../utils/buildPrice';

import type { Ticket } from '../../models/Ticket';
import type { LineUp, LineUpEntry } from '../../models/LineUp';

type MoreInfoProps = {
    description: string,
    lineup: LineUp,
    ticket_types: Array<Ticket>,
    currency: string
};

const MoreInfo = ({ description, lineup, ticket_types, currency }: MoreInfoProps) => {
    const [isOpen, setIsOpen] = useState(false);

    const lineupList = lineup.map((lineupEntry: LineUpEntry) => ({
        primary: lineupEntry.details,
        secondary: lineupEntry.time
    }));

    const ticketsList = ticket_types.map((ticket: Ticket) => ({
        primary: ticket.name,
        secondary: buildPrice(currency, ticket.price.total, 2),
        tertiary: ticket['sold_out']
    }));

    return (
        <div className={styles[isOpen ? 'container-open' : 'container']}>
            <ImportantText>
                More info
                <div className={styles.expand} onClick={() => setIsOpen(!isOpen)}>+</div>
            </ImportantText>
            <Text className={styles.description}>{description}</Text>
            <List title='line up' list={lineupList} />
            <List title='tickets' list={ticketsList} />
        </div>
    );
};

export default MoreInfo