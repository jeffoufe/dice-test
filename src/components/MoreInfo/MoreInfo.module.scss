@import '../../constants/colors';

.container {
    padding: 11px 16px;
    margin-top: 10px;
    background-color: $grey;
    overflow: hidden;
    // height: 19px;
    transition: height 2s;

    &-expanded {
        height: auto;
    }
}

.expand {
    float: right;
    cursor: pointer;
    font-size: 25px;
    font-weight: 500;
}

.description {
    margin-top: 15px;
}