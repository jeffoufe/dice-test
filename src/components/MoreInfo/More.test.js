import MoreInfo from './MoreInfo';
import { shallow } from 'enzyme';
import React from 'react';

describe('MoreInfo', () => {
    const props = {
        lineup: [
            { details: 'artist', time: '8PM' },
            { details: 'artist2', time: '7AM' }

        ],
        ticket_types: [
            { name: 'general', price: { total: 746 }, sold_out: false },
            { name: 'backstage', price: { total: 764 }, sold_out: true },
        ],
        currency: 'GBP',
        description: 'description'
    }

    const wrapper = shallow(
        <MoreInfo {...props} />
    );

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})