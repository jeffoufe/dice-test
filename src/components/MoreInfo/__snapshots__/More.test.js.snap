// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInfo renders correctly 1`] = `
<div
  className="container"
>
  <ImportantText>
    More info
    <div
      className="expand"
      onClick={[Function]}
    >
      +
    </div>
  </ImportantText>
  <Text
    className="description"
  >
    description
  </Text>
  <List
    list={
      Array [
        Object {
          "primary": "artist",
          "secondary": "8PM",
        },
        Object {
          "primary": "artist2",
          "secondary": "7AM",
        },
      ]
    }
    title="line up"
  />
  <List
    list={
      Array [
        Object {
          "primary": "general",
          "secondary": "£7.46",
          "tertiary": false,
        },
        Object {
          "primary": "backstage",
          "secondary": "£7.64",
          "tertiary": true,
        },
      ]
    }
    title="tickets"
  />
</div>
`;
